cpp/src/parquet/column_writer.cc

```diff
@@ -1033,9 +1033,13 @@ void ColumnWriterImpl::BuildDataPageV2(int64_t definition_levels_rle_size,
   // Compress the values if needed. Repetition and definition levels are uncompressed in
   // V2.
   std::shared_ptr<Buffer> compressed_values;
-  if (pager_->has_compressor()) {
+  bool page_is_compressed = false;
+  if (pager_->has_compressor() && values->size() > 0) {
     pager_->Compress(*values, compressor_temp_buffer_.get());
-    compressed_values = compressor_temp_buffer_;
+    if (compressor_temp_buffer_->size() < values->size()) {
+      compressed_values = compressor_temp_buffer_;
+      page_is_compressed = true;
+    }
```

Comment:
aren't we missing to set `compressed_values` on the case  of `values->size` being smaller than the compressed size?
Python tests are currently segfaulting and this is the only thing I can think about without testing and debugging myself.

Reply:
You're right, let me push again this this case